pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -37,6 +39,7 @@ const (
 	ipv6
 	dual
 
+	gkeCNIConfigDir  = "/etc/cni/net.d/10-gke-ptp.conflist"
```

Comment:
This one is tricky. How will we handle the Calico case?

Reply:
Good catch - with Calico there are a few more things to consider:
- CNI file may not be available on netd startup.
- Need to pipe through the Calico CNI name.
- Parsing is a bit different as Calico puts `{ "subnet": "usePodCidr" }` instead of a plain CIDR.

Thought about parsing the CNI template file instead but that may not work as the template file will be removed by Calico pod promptly.

It may be fine to start without supporting this case though.

Reply:
> CNI file may not be available on netd startup.

Not really. install-cni.sh waits for calico's conflist before moving ahead.

Reply:
For reference, Calico's regular conf:
```
{
   "name":"k8s-pod-network",
   "cniVersion":"0.3.1",
   "plugins":[
      {
         "type":"calico",
         "mtu":1460,
         "log_level":"info",
         "log_file_path":"/var/log/calico/cni/cni.log",
         "datastore_type":"kubernetes",
         "nodename":"gke-calico-netd-default-pool-8ce288fc-wb07",
         "ipam":{
            "type":"host-local",
            "subnet":"usePodCidr"
         },
         "policy":{
            "type":"k8s"
         },
         "kubernetes":{
            "kubeconfig":"/etc/cni/net.d/calico-kubeconfig"
         }
      },
      {
         "type":"portmap",
         "capabilities":{
            "portMappings":true
         },
         "snat":true
      },
      {
         "type":"bandwidth",
         "capabilities":{
            "bandwidth":true
         }
      }
   ]
}
```

Direct path conf:
```
{
   "name":"gke-pod-network",
   "cniVersion":"0.3.1",
   "plugins":[
      {
         "type":"calico",
         "mtu":1460,
         "log_level":"warning",
         "log_file_path":"/var/log/calico/cni/cni.log",
         "datastore_type":"kubernetes",
         "nodename":"gke-calico-netd-v6-default-pool-fc507ad4-vmx2",
         "ipam":{
            "type":"host-local",
            "ranges":[
               [
                  {
                     "subnet":"usePodCidr"
                  }
               ],
               [
                  {
                     "subnet":"2600:2d00:4005:c4e0:a80:3a:0:0/112"
                  }
               ]
            ],
            "routes":[
               {
                  "dst":"0.0.0.0/0"
               },
               {
                  "dst":"2001:4860:8040::/42"
               }
            ]
         },
         "policy":{
            "type":"k8s"
         },
         "kubernetes":{
            "kubeconfig":"/etc/cni/net.d/calico-kubeconfig"
         }
      },
      {
         "type":"portmap",
         "capabilities":{
            "portMappings":true
         },
         "snat":true
      },
      {
         "type":"bandwidth",
         "capabilities":{
            "bandwidth":true
         }
      }
   ]
}
```

/etc/cni/net.d/10-calico.conflist